src/core/memory.d

```diff
@@ -899,20 +987,51 @@ void pureFree(void* ptr) @system pure @nogc nothrow
 }
 
 // locally purified for internal use here only
+static import core.stdc.errno;
+
+private enum canLinkToErrno = __traits(getOverloads, core.stdc.errno, "errno").length == 1
+    && __traits(getLinkage, core.stdc.errno.errno) == "C";
 
+version (D_BetterC) @nogc nothrow private @system
```

Comment:
Will this implementation work in non-betterC builds?  Can't we use this implementation for both -betterC and non-betterC builds?

Reply:
It would work. The purpose of the version branch is so outside code that assumes `fakePureErroImpl` exists and tries to link to it will not break.